### PR TITLE
Update pandas 1.1.0

### PR DESCRIPTION
**Pull request by fhoehle**
# Description:

Addressing future warnings `is_categorical` and fix tests/adapt to fixed `assert_series_equal` behavior.

- [x] Changelog entry

### CODE REVIEW

**Comment by fhoehle** [comment]
@marco-neumann-by @lr4d @fjetter please review/comment/merge, thanks.

File: kartothek/io/testing/read.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def use_dataset_factory(request, dates_as_object):
 
 def _strip_unused_categoricals(df):
     for col in df.columns:
-        if pd.api.types.is_categorical(df[col]):
+        if pd.api.types.is_categorical_dtype(df[col]):
```
Comment (marco-neumann-by) [inline kartothek/io/testing/read.py:84]: Is `is_categorical_dtype` included in pandas `0.23.0`?

Comment (fhoehle) [reply kartothek/io/testing/read.py:84]: @marco-neumann-by see: https://github.com/pandas-dev/pandas/blob/v0.23.0/pandas/core/dtypes/common.py#L513

Comment (marco-neumann-by) [reply kartothek/io/testing/read.py:84]: Thanks. `master` is currently broken due to a sphinx bug. I'll need to isolate + report + avoid (likely by constraining sphinx) the bug, but might not have time today. I'll take care of this PR here afterwards, so there is nothing to do from your side.

Comment (lr4d) [reply kartothek/io/testing/read.py:84]: @marco-neumann-by should we add this older-yet-still-supported pandas version to our test matrix?

**Comment by lr4d** [comment]
@fhoehle now that #333 is merged, can you rebase so that the build passes?

File: kartothek/io/testing/read.py (hunk 1 above)
Comment (marco-neumann-by) [reply kartothek/io/testing/read.py:84]: Probably after we've reduced our set of pyarrow versions a bit.

**Pull request merged**